### PR TITLE
New user Registration, Confirmation and session management workflow

### PR DESCRIPTION
**Pull request by danguita**
Connects to #126.

### What does this PR do?

This PR implements the latest specs regarding User registration and session management. In brief, it adds a new Registration and Confirmation workflow, which can be accessed right from the login screen. This updated UI is also providing a way for requesting a password change.

It is also localising and translating any related template by keeping conventions established in the `GobiertoAdmin` module.

#### What is missing?

Once again, I'd need some help with CA translations. I'm keeping them mirrored from ES versions to leave a placeholder to work with 🙏 

### How should this be manually tested?

Just get into the new login screen at http://madrid.gobierto.dev/user/login, and check that:

- [x] Any User can recover her password.
- [x] Any Visitor can register in the new workflow: 1. Ask for email, 2. Ask for the remaining data.
- [x] Any User can start a session with valid credentials.
- [x] Every interaction is localised and translated into the current locale (please note the CA translations are not done yet).
- [x] The UI looks and behaves like the prototype.

Here're a couple of shots on those:

screen shot 2016-12-14 at 14 25 42

screen shot 2016-12-14 at 14 27 36

### CODE REVIEW

**Comment by ferblape** [comment]
I have made a few changes:

1 - include missing Catalan translations, so now the feature is complete
2 - remove duplicated login routes
3 - fix the name of the gender attribute, because when there were validation errors, it wasn't very clear which attribute it was
4 - sorted and updated the years of the year of birth selector

**Comment by ferblape** [comment]
BTW, great job!

File: app/controllers/user/confirmations_controller.rb [hunk 1]
```diff
@@ -52,6 +52,6 @@ def get_user_genders
   end
 
   def get_user_years_of_birth
-    (18.years.ago.year..Date.current.year)
+    (100.years.ago.year..10.years.ago.year).to_a.reverse
```
Comment (danguita) [inline app/controllers/user/confirmations_controller.rb:55]: I do need more sleep 😞

**Comment by danguita** [comment]
Thanks for taking care, go ahead! 👍

**Pull request merged**